sample/src/main/java/com/rd/pageindicatorview/home/HomeActivity.java

```diff
@@ -61,11 +81,17 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     @SuppressWarnings("ConstantConditions")
     private void initViews() {
-        HomeAdapter adapter = new HomeAdapter();
-        adapter.setData(createPageList());
-
-        final ViewPager pager = findViewById(R.id.viewPager);
-        pager.setAdapter(adapter);
+        final Object pager = findViewById(R.id.viewPager);
+        if(pager instanceof ViewPager) {
+            HomePagerAdapter homePagerAdapter = new HomePagerAdapter();
+            homePagerAdapter.setData(createPageList());
+
+            ((ViewPager) pager).setAdapter(homePagerAdapter);
+        } else if(pager instanceof ViewPager2) {{
+            HomeAdapter homeAdapter = new HomeAdapter();
+            homeAdapter.setData(createPageList());
+            ((ViewPager2) pager).setAdapter(homeAdapter);
+        }}
```

Comment:
Supongo que el comentario de @davidgarciaanton venía por aquí.... Up to you si quieres aplicar la mejora :).